redfish/network_protocol.go

```diff
@@ -0,0 +1,292 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package redfish
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+type Protocol struct {
+	// The protocol port
+	Port int64
+	// An indication of whether the protocol is enabled
+	ProtocolEnabled bool
+}
+
+type HTTPS struct {
+	Protocol
+
+	// certificates shall be a link to HTTP certificates resource
+	certificates string
+}
+
+func (https *HTTPS) UnmarshalJSON(b []byte) error {
+	type temp HTTPS
+
+	var t struct {
+		temp
+		Certificates common.Link
+	}
+	err := json.Unmarshal(b, &t)
+	if err != nil {
+		return err
+	}
+	*https = HTTPS(t.temp)
+	https.certificates = t.Certificates.String()
+	return nil
+}
+
+type NTP struct {
+	Protocol
+
+	// Indicates to which user-supplied NTP servers this manager is
+	// subscribed
+	NTPServers []string
+	// The NTP servers supplied by other network protocols to this
+	// manager
+	NetworkSuppliedServers []string
+}
+
+type SNMPAuthenticationProtocol string
+
+const (
+	// Authentication is determined by account settings
+	SNMPAuthAccount SNMPAuthenticationProtocol = "Account"
+	// SNMP community string authentication
+	SNMPAuthCommunityString SNMPAuthenticationProtocol = "CommunityString"
+	// HMAC-MD5-96 authentication
+	SNMPAuthHMAC_MD5 SNMPAuthenticationProtocol = "HMAC_MD5" // nolint
+	// HMAC-SHA-96 authentication
+	SNMPAuthHMAC_SHA96 SNMPAuthenticationProtocol = "HMAC_SHA96" // nolint
+	// HMAC-128-SHA-224 authentication
+	SNMPAuthHMAC128_SHA224 SNMPAuthenticationProtocol = "HMAC128_SHA224" // nolint
+	// HMAC-192-SHA-256 authentication
+	SNMPAuthHMAC192_SHA256 SNMPAuthenticationProtocol = "HMAC192_SHA256" // nolint
+	// HMAC-256-SHA-384 authentication
+	SNMPAuthHMAC256_SHA384 SNMPAuthenticationProtocol = "HMAC256_SHA384" // nolint
+	// HMAC-384-SHA-512 authentication
+	SNMPAuthHMAC384_SHA512 SNMPAuthenticationProtocol = "HMAC384_SHA512" // nolint
+)
+
+type SNMPCommunityAccessMode string
+
+const (
+	// READ-WRITE access mode
+	SNMPCommunityAccessFull SNMPCommunityAccessMode = "Full"
+	// READ-ONLY access mode
+	SNMPCommunityAccessLimited SNMPCommunityAccessMode = "Limited"
+)
+
+type SNMPCommunity struct {
+	// The access level of the SNMP community
+	AccessMode SNMPCommunityAccessMode
+	// The SNMP community string
+	CommunityString string
+	// The name of the SNMP community
+	Name string
+}
+
+type SNMPEncryptionProtocol string
+
+const (
+	// No encryption
+	NoneEncryption SNMPEncryptionProtocol = "None"
+	// Encryption is determined by account settings
+	AccountEncryption SNMPEncryptionProtocol = "Account"
+	// CBC-DES encryption
+	CBC_DES_Encryption SNMPEncryptionProtocol = "CBC_DES" // nolint
+	// CFB128-AES-128 encryption
+	CFB128_AES128_Encryption SNMPEncryptionProtocol = "CFB128_AES128" // nolint
+)
+
+type EngineId struct {
```

Comment:
Should be `EngineID`.

As well as all the other `*Id` names in here.

Reply:
Done!